src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7845,14 +7847,42 @@ private GotoStatementSyntax ParseGotoStatement()
 
         private IfStatementSyntax ParseIfStatement()
         {
-            Debug.Assert(this.CurrentToken.Kind == SyntaxKind.IfKeyword);
-            var @if = this.EatToken(SyntaxKind.IfKeyword);
-            var openParen = this.EatToken(SyntaxKind.OpenParenToken);
-            var condition = this.ParseExpressionCore();
-            var closeParen = this.EatToken(SyntaxKind.CloseParenToken);
-            var statement = this.ParseEmbeddedStatement();
-            var elseClause = ParseElseClauseOpt();
+            Debug.Assert(this.CurrentToken.Kind == SyntaxKind.IfKeyword || this.CurrentToken.Kind == SyntaxKind.ElseKeyword);
+
+            SyntaxToken @if;
+            SyntaxToken openParen;
+            ExpressionSyntax condition;
+            SyntaxToken closeParen;
+            StatementSyntax statement;
+            ElseClauseSyntax elseClause;
 
+            if (this.CurrentToken.Kind == SyntaxKind.IfKeyword)
+            {
+                @if = this.EatToken(SyntaxKind.IfKeyword);
+                openParen = this.EatToken(SyntaxKind.OpenParenToken);
+                condition = this.ParseExpressionCore();
+                closeParen = this.EatToken(SyntaxKind.CloseParenToken);
+                statement = this.ParseEmbeddedStatement();
+                elseClause = this.ParseElseClauseOpt();
+            }
+            else
+            {
+                elseClause = this.ParseElseClauseOpt();
+                elseClause = this.AddErrorToFirstToken(elseClause, ErrorCode.ERR_SyntaxError, SyntaxFacts.GetText(SyntaxKind.IfKeyword));
+
+                // synthesize missing tokens for "if (true) { }":
+                @if = SyntaxToken.CreateMissing(SyntaxKind.IfKeyword, null, null);
+                openParen = SyntaxToken.CreateMissing(SyntaxKind.OpenParenToken, null, null);
+                condition = this._syntaxFactory.LiteralExpression(
+                    SyntaxKind.TrueLiteralExpression, 
+                    SyntaxToken.CreateMissing(SyntaxKind.TrueKeyword, null, null));
+                closeParen = SyntaxToken.CreateMissing(SyntaxKind.CloseParenToken, null, null);
+                statement = _syntaxFactory.Block(
+                    SyntaxToken.CreateMissing(SyntaxKind.OpenBraceToken, null, null),
+                    default,
+                    SyntaxToken.CreateMissing(SyntaxKind.CloseBraceToken, null, null));
+            }
```

Comment:
Please add some parsing tests to demonstrate the shape of the parse tree after the change. Look at `StatementParsingTests.ParsePrivate` as an example.

Note that the `M` method will print out the correct code, but you first have to uncomment `#define PARSING_TESTS_DUMP
` in `src\Compilers\CSharp\Test\Syntax\Parsing\ParsingTests.cs`.